stdlib/src/locale.rs

```diff
@@ -83,9 +81,8 @@ mod _locale {
         unsafe {
             let string_cstr = CString::new(string.as_str()).map_err(|e| e.to_pyexception(vm))?;
             let string_ptr = CStr::as_ptr(&string_cstr);
-            let empty_str = CString::new("").unwrap();
-            let result_ptr = CStr::as_ptr(&empty_str) as *mut c_char;
-            libc::strxfrm(result_ptr, string_ptr, libc::strlen(string_ptr) + 1);
+            let result_ptr = ptr::null_mut();
+            libc::strxfrm(result_ptr, string_ptr, string.char_len() + 1);
```

Comment:
the return value is length. you can create buffer with the given length and call strxfrm again.

Reply:
Wait! So strxfrm is called 2 times?